### PR TITLE
Get rid of the NFS vs JIANT environment variable distinction

### LINKED ISSUE #613: ${NFS_PROJECT_PREFIX} and ${JIANT_PROJECT_PREFIX}
Do we need two separate set of environment variables?
We also have ${NFS_DATA_DIR} and ${JIANT_DATA_DIR}. I don't know about potential users of jiant, at least for me, it's pretty confusing.

### PR DESCRIPTION
**Pull request by sleepinyourhat**
This distinction is confusing in most cases, and it's still easy to recreate it with custom paths/configs.

Resolves #613 

@iftenney - Mind reviewing?

### CODE REVIEW

**Comment by sleepinyourhat** [comment]
@pep8speaks pep8ify

File: config/edgeprobe_existing.conf [hunk 1]
```diff
@@ -9,9 +9,6 @@
 
 // Override paths from params.conf, since these might point to paths on a
 // different system.
-global_ro_exp_dir = "/nfs/jsalt/share/exp/default"
-project_dir = ${JIANT_PROJECT_PREFIX}
-data_dir = ${JIANT_DATA_DIR}  // required - should point to data on NFS.
```
Comment (iftenney) [inline config/edgeprobe_existing.conf]: How is this configured now? Directly with env vars?

File: main.py [hunk 2]
```diff
@@ -34,7 +34,7 @@
 def handle_arguments(cl_arguments):
     parser = argparse.ArgumentParser(description='')
     # Configuration files
-    parser.add_argument('--config_file', '-c', type=str, nargs="+",
+    parser.add_argument('--config_file', '-c', type=str, nargs="+", default="config/defaults.conf",
```
Comment (iftenney) [inline main.py]: If setting this default, might want to enforce that `main.py` is run from the main jiant/ directory as well?

File: config/edgeprobe_existing.conf (hunk 1 above)
Comment (sleepinyourhat) [reply config/edgeprobe_existing.conf]: Oops—I missed that this wasn't inheriting from defaults.conf.

File: main.py (hunk 2 above)
Comment (sleepinyourhat) [reply main.py]: That sounds pretty non-standard, no? Wouldn't impact many use cases, since you'll nearly always want to use a custom conf.

**Comment by sleepinyourhat** [comment]
Ready for another look.

**Comment by iftenney** [comment]
Fix conflicts, and I can take another look?

File: README.md [hunk 3]
```diff
@@ -2,8 +2,6 @@
 
 [![CircleCI](https://circleci.com/gh/nyu-mll/jiant/tree/master.svg?style=svg)](https://circleci.com/gh/nyu-mll/jiant/tree/master)
 
-This repo contains the `jiant` sentence representation learning toolkit created at the [2018 JSALT Workshop](https://www.clsp.jhu.edu/workshops/18-workshop/) by the [General-Purpose Sentence Representation Learning](https://jsalt18-sentence-repl.github.io/) team. It is an extensible platform meant to make it easy to run experiments that involve multitask and transfer learning across sentence-level NLP tasks.
```
Comment (sleepinyourhat) [inline README.md]: I think this was added through someone else's botched merge...

**Comment by sleepinyourhat** [comment]
Done.

**Pull request merged**